File: Solutions/String/13 Roman to Integer.cs
//roman to int
public class Solution13
{
    public int RomanToInt(string s)
    {
        Dictionary<char, int> map = new Dictionary<char, int>();
        map.Add('I', 1);
        map.Add('V', 5);
        map.Add('X', 10);
        map.Add('L', 50);
        map.Add('C', 100);
        map.Add('D', 500);
        map.Add('M', 1000);

        int res = map[s[s.Length - 1]];

        for (int i = s.Length - 2; i >= 0; i--)
        {
            if (map[s[i]] < map[s[i + 1]])
            {
                res -= map[s[i]];
            }
            else
            {
                res += map[s[i]];
            }
        }

        return res;
    }
}

File: 13-roman-to-integer/13-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        int l = s.size();
        int res=0;
        map<int,int> m;
        m['I']=1;
        m['V']=5;
        m['X']=10;
        m['L']=50;
        m['C']=100;
        m['D']=500;
        m['M']=1000;
        res=m[s[l-1]];
        for(int i=l-2;i>=0;i--){
            if(m[s[i]]<m[s[i+1]]){
                res-=m[s[i]];
            }else{
                res+=m[s[i]];
            }
        }
        return res;
    }
};